### PR TITLE
Add new event overlay features and use cases

### PR DESCRIPTION
**Pull request by estherk15**
### What does this PR do?
- Add new features of the Event Overlay
- Use cases
- Link to Timeseries and Heat Maps

### Motivation
[DOCS-5041](https://datadoghq.atlassian.net/browse/DOCS-5041)

### Additional Notes
**Do not merge** Pending review from Product team.

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

[DOCS-5041]: https://datadoghq.atlassian.net/browse/DOCS-5041?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: content/en/dashboards/querying/_index.md [hunk 1]
```diff
@@ -210,7 +210,17 @@ If your level of detail is resource or span, some widget types also require you
 
 ### Event overlays
 
-View event correlations by using the **Event Overlays** section in the graphing editor. In the search field, enter any text or structured search query. For details on searching, see the Datadog [Event Query Language][14].
+View event correlations by using the **Event Overlays** section in the graphing editor for the [Timeseries][15] visualization. In the search field, enter any text or structured search query. For details on searching, see the Datadog [Event Query Language][14].
```
Comment (drichards-87) [inline content/en/dashboards/querying/_index.md:213]: I think the link for Event Query Language is broken... Can you confirm?

**Review by drichards-87 — CHANGES_REQUESTED** [review]
Looks good. I just had a question re: one of the links.

File: content/en/dashboards/querying/_index.md (hunk 1 above)
Comment (estherk15) [reply content/en/dashboards/querying/_index.md:213]: Ooh, good catch, yea this section doesn't seem to exist

**Comment by omar-nema** [comment]
thank you! looks good, but i think this paragraph on querying page could be improved

`You can add events from related systems to add more context to your graph, such as GitHub commits, Jenkins deploys, and Docker creation events. Click Add Event Overlay in the Event Overlays section and enter a query to display those events. The event overlay supports all data sources. This allows for easier correlation between business events and data from any Datadog product.`

I think the first and last sentences here are related to context, but the middle two are more about using the feature. Recommend switching this up and consolidating so it flows better

**Pull request merged**